docs/dev/oauth_for_data_integrations.md

```diff
@@ -0,0 +1,137 @@
+
+   In the authorization code grant flow, you receive an authorization code and refresh token, then exchange the code for an access token that can be used to access the data you want to pull from Datadog. 
+
+   For more information about implementing the OAuth protocol with Datadog, see [Datadog OAuth2][1]. For more information about building and publishing an integration, see the [Integrations developer documentation][5].
+
+6. Test the OAuth protocol by clicking **Test Authorization** on your client's details page. This directs you to the onboarding URL and starts the authorization flow that a customer takes. 
+
+    Before your client is published, only members of your Datadog organization can authorize against your client during testing. 
+
+7. To validate that OAuth is working properly, make a request to the `marketplace_create_api` endpoint with your token in the headers of the request. 
+
+   If successful, this request returns an API key that you should securely save so you can use it to submit data into Datadog on behalf of the user. **You cannot access this API key value again after the initial request response**.
+
+8. Test that your OAuth client can work across multiple [Datadog sites][8] by kicking off authorization from a non-US1 Datadog account:
+   - If you do not have access to a sandbox account on a different site, contact `marketplace@datadog.com`. 
+   - Export your app manifest from the account in the *original* Datadog site by navigating to the app you've created in the Developer Platform, clicking the gear icon on the top right, and clicking **Export App Manifest**. 
+   - In your account on the *new* Datadog site, navigate to the Developer Platform and import your app manifest from Step 2.
+   - After successfully importing your manifest, navigate to the **OAuth & Permissions** tab to find your OAuth client, along with its client ID and client secret. Use these credentials to test authorization from this site. 
+
+9. Test any additional scopes that you’ve requested access for.
+
+### Publish the OAuth client
+
+In order to publish an OAuth client, you first need to open a pull request for your data integration in either the [`integrations-extras`][5] or [Marketplace][6] GitHub repositories. 
+
+As a part of your pull request, update your README file with an **uninstallation** section under `## Setup` that includes the following instructions (along with any custom instructions you would like to add):
+
+- To ensure this integration is fully uninstalled, revoke authorization in the Configure tab of this tile or in [OAuth Apps][9] within Organization Settings in Datadog. 
+- Additionally, ensure that all API keys associated with this integration have been disabled by searching for the integration name on the [API Keys management page][10].
```

Comment:
@alai97 This is text that the user should copy and paste into their readme file. Is there any kind of UI Component we can use to represent this?

Reply:
Good catch!